code/game/gamemodes/miniantags/revenant/revenant_abilities.dm

```diff
@@ -271,50 +271,135 @@
-		new /obj/effect/temp_visual/revenant(human.loc)
-		human.emp_act(EMP_HEAVY)
-	for(var/obj/thing in T)
-		if(istype(thing, /obj/machinery/dominator) || istype(thing, /obj/machinery/power/apc) || istype(thing, /obj/machinery/power/smes)) //Doesn't work on dominators, SMES and APCs, to prevent kekkery
-			continue
-		if(prob(20))
-			if(prob(50))
-				new /obj/effect/temp_visual/revenant(thing.loc)
-			thing.emag_act(null)
-		else
-			if(!istype(thing, /obj/machinery/clonepod)) //I hate everything but mostly the fact there's no better way to do this without just not affecting it at all
-				thing.emp_act(EMP_HEAVY)
-	for(var/mob/living/silicon/robot/S in T) //Only works on cyborgs, not AI
-		playsound(S, 'sound/machines/warning-buzzer.ogg', 50, 1)
-		new /obj/effect/temp_visual/revenant(S.loc)
-		S.spark_system.start()
-		S.emp_act(EMP_HEAVY)
+/obj/effect/proc_holder/spell/targeted/revenant/madness/proc/mind_sounds()
+	playsound(src, 'sound/magic/divulge_ending.ogg', 50, 1, -1)
+
+/obj/effect/proc_holder/spell/targeted/revenant/animate_bone
+	name = "Animate Bone"
+	desc = "Exhumes the skeleton from it's host, as long as the body has been drained. It will hunt down the living."
+	charge_max = 300
+	range = 1
+	include_user = 0
+	reveal = 0 //How long it reveals the revenant in deciseconds
+	stun = 0 //How long it stuns the revenant in deciseconds
+	unlock_amount = 74
+	cast_amount = 30
+	action_icon_state = "animate_bone"
+
+/obj/effect/proc_holder/spell/targeted/revenant/animate_bone/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(attempt_cast(user))
+		for(var/mob/living/M in targets)
+			if(!(M in user.drained_mobs))
+				to_chat(user, "<span class='revenwarning'>Harvest the soul first!</span>")
+				return FALSE
+			to_chat(user, "<span class='revenboldnotice'>Ah yes, this will do nicely! You charge up your necrotic powers...</span>")
+			if(!do_after(user, 30, 0, M))
+				return FALSE
+			user.reveal(66)
+			user.stun(66)
+			M.visible_message("<span class='warning'>[M] starts shaking violently!</span>", \
+				  "<span class='userdanger'>You lock up, but your bones do not! THEY'RE TRYING TO GET OUT!</span>")
+			M.Knockdown(66)
+			M.Jitter(66)
+			M.adjustBruteLoss(30) //i'm pretty sure your skeleton trying to get out hurts, but i'm no doctor.
+			var/datum/beam/B = user.Beam(M,icon_state="animate",time=INFINITY)
+			if(!do_after(user, 66, 0, M))
+				to_chat(user, "<span class='revenwarning'>The animation has been broken!</span>")
+				qdel(B)
+				return FALSE
+			qdel(B)
+			M.visible_message("<span class='userdanger'>[M]'s skeleton explodes out of them in a shower of gore!</span>")
+			var/mob/living/simple_animal/hostile/skeleton/revenant/S
+			S = new(M.loc)
+			S.name = "[M]'s haunted remains"
+			if(length(M.logging[INDIVIDUAL_SAY_LOG]))
+				S.lastthingtheysaid = M.logging[INDIVIDUAL_SAY_LOG][M.logging[INDIVIDUAL_SAY_LOG].len]
```

Comment:
so this would just be `S.speak += M.logging[INDIVIDUAL_SAY_LOG][M.logging[INDIVIDUAL_SAY_LOG].len]`

Reply:
instead of telling me what to do also showing how i would do it. THANK YOU SO MUCH.